src/main/cpp/win_fsnotifier.cpp

```diff
@@ -51,99 +176,96 @@ void handlePathChanged(watch_details_t *details, FILE_NOTIFY_INFORMATION *info)
     } else if (info->Action == FILE_ACTION_MODIFIED) {
         type = FILE_EVENT_MODIFIED;
     } else {
-        wprintf(L"~~~~ Unknown event 0x%x for %ls\n", info->Action, changedPath);
+        wprintf(L"~~~~ Unknown event 0x%x for %ls\n", info->Action, changedPath.c_str());
         type = FILE_EVENT_UNKNOWN;
     }
 
-    bool watching = false;
-    for (int i = 0; i < details->watchedPathCount; i++) {
-        wchar_t* watchedPath = details->watchedPaths[i];
-        if (wcsncmp(watchedPath, changedPath, wcslen(watchedPath)) == 0) {
-            watching = true;
-            break;
-        }
-    }
-    if (!watching) {
-        wprintf(L"~~~~ Ignoring %ls (root is not watched)\n", changedPath);
-        return;
+    server->reportEvent(type, changedPath);
+}
+
+void Server::startWatching(wchar_t *path) {
+    WatchPoint* watchPoint = new WatchPoint(this, wstring(path));
```

Comment:
Can you do anything to avoid manual new/delete, e.g. containers, or even unique/shared_ptr?

Reply:
I guess I could if I knew how C++ worked. :) Care to enlighten about my options here?